### PR TITLE
Set up JS pipelines

### PR DESCRIPTION
**Pull request by arpanlaha**
Related to #54 - attempt at creating pipeline needed for #53.

To be honest, I have no idea if I'm doing this right but I want to have the following functionality:

- Build - `npm run build` and `npm pack`
- Analyze - `npm run lint` and `npm audit`
- Test - `npm test`

The project defined in #53 will reside in `src/ts/eslint-plugin-azure-sdk`, which I access by `cd`ing in through a script block, but I'm not sure if this is the correct way to do it or how to make it modular to other projects in `src/ts`.

@Azure/azure-sdk-eng

### CODE REVIEW

File: .azure-pipelines/js.yml [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        displayName: "Run lint"
+
+      - script: |
+          npm audit
+        condition: and(succeeded(), eq(variables['RunNpmAudit'], 'true'))
+        displayName: "Audit packages"
+
+  - job: "Test"
+
+    strategy:
```
Comment (KarishmaGhiya) [inline .azure-pipelines/js.yml]: @arpanlaha Let's remove this matrix and test only on one vm (ubuntu) and one node version specified above in the variables `NodeVersion: "10.x"` 
For reference: https://github.com/Azure/azure-sdk-tools/blob/master/.azure-pipelines/dotnet.yml#L38

Comment (arpanlaha) [reply .azure-pipelines/js.yml]: done!

**Comment by mitchdenny** [comment]
I would be inclined to lift this module and put it under ```tools/eslint-plugin-azure-sdk```. And then name your Azure Pipelines YAML file ```ci.yml``` and drop it under that directory. I've got some changes coming in this pipeline which will make it such that the _azure-sdk-tools_ repo builds in a similar way to the other mono repos and doing what I'm proposing here would snap us towards that model.

**Comment by KarishmaGhiya** [comment]
@arpanlaha the changes you made to remove the matrix look good. You can incorporate the suggestions and address the questions made by Mitch :)

**Comment by arpanlaha** [comment]
> Suggestions:
> 
> 1. Relocate the package itself under `tools/eslint-plugin-azure-sdk`.
> 2. Rename the `js.yml` file to `ci.yml` and move it under `tools/eslint-plugin-azure-sdk`.
> 
> How will this package be consumed in the JS repo - will you be publishing it to NPM and consuming it as a dev dependency?

@mitchdenny added your suggestions! This will be published to NPM under the `ts-common` scope and is intended to be used as a dev dependency by all the JS/TS service packages to streamline the guideline review process. Ideally all that would be required there would be to add the dev dependency and add 2-3 lines to the package's `eslintrc` to import the plugin and its recommended configuration.

**Pull request merged**